### PR TITLE
ant: avoid inserting diodes for drivers

### PR DESCRIPTION
**Pull request by eder-matheus**

### CODE REVIEW

**Comment by eder-matheus** [comment]
@maliberty I've started a CI with the public designs to check the impact of it in the final violation count. I expect no changes in it, but lets see how it goes.

**Review by maliberty — APPROVED** [review]

**Comment by eder-matheus** [comment]
@maliberty I've updated this PR to return only the pin with violation in each ant::Violation object. This way, drivers will not have diodes near them.
The crash we had was due to a clear in a std::map in multithreading. I moved this clear to a better place now.

File: src/ant/src/AntennaChecker.cc [hunk 1]
```diff
@@ -772,7 +774,12 @@ int AntennaChecker::checkGates(odb::dbNet* db_net,
   for (const auto& [node, layer_to_node] : gate_info) {
     bool pin_has_violation = false;
 
-    std::string pin_name = fmt::format("  Pin: {}", node);
+    odb::dbMTerm* mterm = node->getMTerm();
+    std::string node_name = fmt::format("  {}/{} ({})",
+                                        node->getInst()->getConstName(),
+                                        mterm->getConstName(),
+                                        mterm->getMaster()->getConstName());
+    std::string pin_name = fmt::format("  Pin: {}", node_name);
```
Comment (maliberty) [inline src/ant/src/AntennaChecker.cc:782]: Why not combine these into "  Pin:   {}/{} ({})" ...

**Review by maliberty — APPROVED** [review]
Does this need a secure CI?

**Comment by eder-matheus** [comment]
> Does this need a secure CI?

CI is passing.

**Pull request merged**